Library/Homebrew/test/test_sandbox.rb

```diff
@@ -0,0 +1,25 @@
+require "testing_env"
+require "sandbox"
+
+exit 0 unless Sandbox.available?
+
+class SandboxTest < Homebrew::TestCase
+  def test_allow_write
+    s = Sandbox.new
+    testpath = Pathname.new(TEST_TMPDIR)
+    foo = testpath/"foo"
+    s.allow_write "#{testpath}", :type => :subpath
+    s.exec "touch", foo
```

Comment:
Tests that mutate the filesystem should roll back the changes (this prevents order dependencies among tests)